src/components/infinite-scroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * From https://github.com/jaredpalmer/react-simple-infinite-scroll
+ * Updated to accept a scroll container React ref as the "window"
+ */
+import * as React from 'react';
+import throttleFn from 'lodash/throttle';
+
+export type Props = {
+    /** Components to render, should include entities/loading states/end message */
+    children: React.ReactNode;
+    /** Does the resource have more entities */
+    hasMore: boolean;
+    /** Are more entities being loaded */
+    isLoading: boolean;
+    /** Callback to load more entities */
+    onLoadMore: () => void;
+    /** React ref of the scroll container.
+     * Used to listen to scroll events and calculate onLoadMore calls.
+     * Set useWindow to true to use the window as scroll container.
+     */
+    scrollContainerNode?: HTMLElement;
+    /** Scroll threshold */
+    threshold?: number;
+    /** Throttle rate */
+    throttle?: number;
+    /** Set useWindow to true to use the window as scroll container. If set to true, will ignore scrollContainerRef. */
+    useWindow?: boolean;
+};
+
+function InfiniteScroll({
+    children,
+    hasMore,
+    isLoading,
+    onLoadMore,
+    scrollContainerNode,
+    threshold = 100,
+    throttle = 64,
+    useWindow = false,
+}: Props) {
+    const sentinelRef = React.createRef<HTMLDivElement>();
+
+    function getContainer(): EventTarget | undefined {
+        return useWindow ? window : scrollContainerNode;
+    }
```

Comment:
Would it work to do `const container = useWindow ? window : scrollContainerNode;` here so that `getContainer()` doesn't need to be called in `addEventListeners` and `removeEventListeners`?

Reply:
yeah, I'll change that